### PR TITLE
Change list column_names in select_from_values to column that can be …

### LINKED ISSUE #137: Package execution fails if exposures are missed in the project
### Describe the bug

```
dbt build --select package:dbt_project_evaluator
```
dbt statement is failing if there is no metrics and/or exposures in the project.
```
15:07:15  Completed with 1 error and 4 warnings:
15:07:15  
15:07:15  Database Error in model int_direct_relationships (models/marts/core/int_direct_relationships.sql)
15:07:15    No matching signature for operator = for argument types: STRING, INT64. Supported signature: ANY = ANY at [60:12]
15:07:15    compiled SQL at target/run/dbt_project_evaluator/models/marts/core/int_direct_relationships.sql
```

In Bigquery views `NULL` values get `INTEGER` data type if none was provided explicitly
```
# View statement is post formatted
SELECT
  *
FROM
  UNNEST([ 
    STRUCT( 
      NULL AS unique_id,
      NULL AS name,
      NULL AS resource_type,
      NULL AS file_path,
      NULL AS is_described,
      NULL AS exposure_type,
      NULL AS maturity,
      NULL AS package_name,
      NULL AS url ) ])
WHERE
  unique_id IS NOT NULL
``` 
And as a result we're getting all fields been integer
image

### Steps to reproduce

Execute `dbt build --select package:dbt_project_evaluator` in a project without exposures

### Expected results

dbt statement executed successfully

### Actual results

Model `int_direct_relationships` is failing.
```
15:07:15  Completed with 1 error and 4 warnings:
15:07:15  
15:07:15  Database Error in model int_direct_relationships (models/marts/core/int_direct_relationships.sql)
15:07:15    No matching signature for operator = for argument types: STRING, INT64. Supported signature: ANY = ANY at [60:12]
15:07:15    compiled SQL at target/run/dbt_project_evaluator/models/marts/core/int_direct_relationships.sql
```
Compiled `int_direct_relationships.sql`
```


  create or replace table `***`.`***`.`int_direct_relationships`
  
  
  OPTIONS()
  as (
    -- one record for each resource in the graph and its direct parent
with 

all_graph_resources as (
    select
        resource_id,
        resource_name,
        resource_type,
        file_path,
        directory_path, 
        file_name,
        model_type,
        source_name 
    from `***`.`***`.`int_all_graph_resources`
),

direct_model_relationships as (
    select  
        resource_id,
        direct_parent_id
    from `***`.`***`.`stg_node_relationships`
),

direct_exposure_relationships as (
    select  
        resource_id,
        direct_parent_id
    from `***`.`***`.`stg_exposure_relationships`
),

direct_metrics_relationships as (
    select  
        resource_id,
        direct_parent_id
    from `***`.`***`.`stg_metric_relationships`
),

-- for all resources in the graph, find their direct parent
direct_relationships as (
    select
        all_graph_resources.*,
        CASE 
            WHEN all_graph_resources.resource_type = 'source' THEN NULL
            WHEN all_graph_resources.resource_type = 'exposure' THEN exposures.direct_parent_id
            WHEN all_graph_resources.resource_type IN ('model', 'snapshot', 'test') THEN models.direct_parent_id
            ELSE NULL
        END AS direct_parent_id
    from all_graph_resources
    left join direct_model_relationships as models
        on all_graph_resources.resource_id = models.resource_id
    left join direct_exposure_relationships as exposures
        on all_graph_resources.resource_id = exposures.resource_id   # Line that fails
),

final as (
    select
        to_hex(md5(cast(coalesce(cast(resource_id as string), '') || '-' || coalesce(cast(direct_parent_id as string), '') as string))) as unique_id,
        *
    from direct_relationships
)

select * from final
  );
 
```

### System information
**The contents of your `packages.yml` file:**
```
packages:
  - git: "git@github.com:zendesk/dbt_zendesk_utils.git"
    revision: 0.1.5
    subdirectory: "src"
  - package: re-data/re_data
    version: [">=0.9.0", "<0.10.0"]
  - git: "https://github.com/dbt-labs/dbt-project-evaluator.git"
    revision: v0.1.0
```

**Which database are you using dbt with?**
- [ ] postgres
- [ ] redshift
- [x] bigquery
- [ ] snowflake
- [ ] other (specify: ____________)

**The output of `dbt --version`:**
```
Core:
  - installed: 1.1.0
  - latest:    1.1.0 - Up to date!

Plugins:
  - bigquery: 1.1.0 - Up to date!
```

### Are you interested in contributing the fix?

Yes, I'd like to contribute with some guidance

### PR DESCRIPTION
**Pull request by VasiliiSurov**
…a column_name or tuple (column_name, data_type)

This is a:
- [x] bug fix PR with no breaking changes
- [ ] new functionality

## Link to Issue 

Closes #137 

## Description & motivation

Changing column_names in `select_from_values` to columns, which can be either column_name (by default) or tuple (column_name, data type). If `values` is empty these data types are used to define correct type in empty model, by default column name will be `string`.

## Integration Test Screenshot

```
(dbt) sova@pop-os:~/IdeaProjects/dbt-project-evaluator/integration_tests$ dbt build
01:28:08  Running with dbt=1.1.1
01:28:08  Found 45 models, 48 tests, 0 snapshots, 0 analyses, 466 macros, 0 operations, 16 seed files, 5 sources, 1 exposure, 1 metric
01:28:08  
01:28:09  Concurrency: 8 threads (target='dev')
01:28:09  
01:28:09  1 of 95 START table model snadbox.stg_exposure_relationships ................... [RUN]
01:28:09  2 of 95 START table model snadbox.stg_exposures ................................ [RUN]
01:28:09  3 of 95 START table model snadbox.stg_metric_relationships ..................... [RUN]
01:28:09  4 of 95 START table model snadbox.stg_metrics .................................. [RUN]
01:28:09  5 of 95 START view model snadbox.stg_naming_convention_folders ................. [RUN]
01:28:09  6 of 95 START view model snadbox.stg_naming_convention_prefixes ................ [RUN]
01:28:09  7 of 95 START table model snadbox.stg_node_relationships ....................... [RUN]
01:28:09  8 of 95 START table model snadbox.stg_nodes .................................... [RUN]
01:28:10  5 of 95 OK created view model snadbox.stg_naming_convention_folders ............ [OK in 1.19s]
01:28:10  9 of 95 START table model snadbox.stg_sources .................................. [RUN]
01:28:10  6 of 95 OK created view model snadbox.stg_naming_convention_prefixes ........... [OK in 1.23s]
01:28:10  10 of 95 START seed file snadbox.test_fct_direct_join_to_source ................ [RUN]
01:28:12  1 of 95 OK created table model snadbox.stg_exposure_relationships .............. [CREATE TABLE (2.0 rows, 0 processed) in 2.57s]
01:28:12  11 of 95 START seed file snadbox.test_fct_documentation_coverage ............... [RUN]
01:28:12  2 of 95 OK created table model snadbox.stg_exposures ........................... [CREATE TABLE (1.0 rows, 0 processed) in 2.59s]
01:28:12  12 of 95 START seed file snadbox.test_fct_missing_primary_key_tests ............ [RUN]
01:28:12  4 of 95 OK created table model snadbox.stg_metrics ............................. [CREATE TABLE (1.0 rows, 0 processed) in 2.58s]
01:28:12  13 of 95 START seed file snadbox.test_fct_model_directories .................... [RUN]
01:28:12  3 of 95 OK created table model snadbox.stg_metric_relationships ................ [CREATE TABLE (1.0 rows, 0 processed) in 2.63s]
01:28:12  14 of 95 START seed file snadbox.test_fct_model_fanout ......................... [RUN]
01:28:12  7 of 95 OK created table model snadbox.stg_node_relationships .................. [CREATE TABLE (143.0 rows, 0 processed) in 2.91s]
01:28:12  15 of 95 START seed file snadbox.test_fct_model_naming_conventions ............. [RUN]
01:28:12  8 of 95 OK created table model snadbox.stg_nodes ............................... [CREATE TABLE (109.0 rows, 0 processed) in 2.94s]
01:28:12  16 of 95 START seed file snadbox.test_fct_multiple_sources_joined .............. [RUN]
01:28:13  9 of 95 OK created table model snadbox.stg_sources ............................. [CREATE TABLE (5.0 rows, 0 processed) in 2.28s]
01:28:13  17 of 95 START seed file snadbox.test_fct_rejoining_of_upstream_concepts ....... [RUN]
01:28:14  10 of 95 OK loaded seed file snadbox.test_fct_direct_join_to_source ............ [INSERT 2 in 3.86s]
01:28:14  18 of 95 START seed file snadbox.test_fct_root_models .......................... [RUN]
01:28:15  12 of 95 OK loaded seed file snadbox.test_fct_missing_primary_key_tests ........ [INSERT 11 in 3.62s]
01:28:15  19 of 95 START seed file snadbox.test_fct_source_directories ................... [RUN]
01:28:15  13 of 95 OK loaded seed file snadbox.test_fct_model_directories ................ [INSERT 6 in 3.66s]
01:28:15  20 of 95 START seed file snadbox.test_fct_source_fanout ........................ [RUN]
01:28:15  14 of 95 OK loaded seed file snadbox.test_fct_model_fanout ..................... [INSERT 1 in 3.64s]
01:28:15  21 of 95 START seed file snadbox.test_fct_staging_dependent_on_staging ......... [RUN]
01:28:15  11 of 95 OK loaded seed file snadbox.test_fct_documentation_coverage ........... [INSERT 1 in 3.72s]
01:28:15  22 of 95 START seed file snadbox.test_fct_test_coverage ........................ [RUN]
01:28:16  16 of 95 OK loaded seed file snadbox.test_fct_multiple_sources_joined .......... [INSERT 1 in 3.72s]
01:28:16  23 of 95 START seed file snadbox.test_fct_test_directories ..................... [RUN]
01:28:16  15 of 95 OK loaded seed file snadbox.test_fct_model_naming_conventions ......... [INSERT 4 in 3.79s]
01:28:16  24 of 95 START seed file snadbox.test_fct_undocumented_models .................. [RUN]
01:28:16  17 of 95 OK loaded seed file snadbox.test_fct_rejoining_of_upstream_concepts ... [INSERT 1 in 3.82s]
01:28:16  25 of 95 START seed file snadbox.test_fct_unused_sources ....................... [RUN]
01:28:18  18 of 95 OK loaded seed file snadbox.test_fct_root_models ...................... [INSERT 1 in 3.73s]
01:28:18  26 of 95 START test not_null_stg_naming_convention_folders_folder_name ......... [RUN]
01:28:19  19 of 95 OK loaded seed file snadbox.test_fct_source_directories ............... [INSERT 1 in 3.64s]
01:28:19  27 of 95 START test unique_stg_naming_convention_folders_folder_name ........... [RUN]
01:28:19  20 of 95 OK loaded seed file snadbox.test_fct_source_fanout .................... [INSERT 2 in 3.65s]
01:28:19  21 of 95 OK loaded seed file snadbox.test_fct_staging_dependent_on_staging ..... [INSERT 1 in 3.64s]
01:28:19  28 of 95 START test not_null_stg_naming_convention_prefixes_unique_id .......... [RUN]
01:28:19  29 of 95 START test unique_stg_naming_convention_prefixes_unique_id ............ [RUN]
01:28:19  22 of 95 OK loaded seed file snadbox.test_fct_test_coverage .................... [INSERT 1 in 3.66s]
01:28:19  30 of 95 START test unique_stg_model_1_id ...................................... [RUN]
01:28:19  26 of 95 PASS not_null_stg_naming_convention_folders_folder_name ............... [PASS in 1.16s]
01:28:19  31 of 95 START test not_null_stg_model_2_id .................................... [RUN]
01:28:20  24 of 95 OK loaded seed file snadbox.test_fct_undocumented_models .............. [INSERT 11 in 3.82s]
01:28:20  32 of 95 START test unique_stg_model_2_id ...................................... [RUN]
01:28:20  23 of 95 OK loaded seed file snadbox.test_fct_test_directories ................. [INSERT 2 in 3.89s]
01:28:20  33 of 95 START test not_null_stg_exposure_relationships_unique_id .............. [RUN]
01:28:20  28 of 95 PASS not_null_stg_naming_convention_prefixes_unique_id ................ [PASS in 1.05s]
01:28:20  34 of 95 START test unique_stg_exposure_relationships_unique_id ................ [RUN]
01:28:20  31 of 95 PASS not_null_stg_model_2_id .......................................... [PASS in 1.02s]
01:28:20  30 of 95 PASS unique_stg_model_1_id ............................................ [PASS in 1.06s]
01:28:20  35 of 95 START test not_null_stg_exposures_unique_id ........................... [RUN]
01:28:20  27 of 95 PASS unique_stg_naming_convention_folders_folder_name ................. [PASS in 1.19s]
01:28:20  36 of 95 START test unique_stg_exposures_unique_id ............................. [RUN]
01:28:20  37 of 95 START test not_null_stg_metrics_unique_id ............................. [RUN]
01:28:20  29 of 95 PASS unique_stg_naming_convention_prefixes_unique_id .................. [PASS in 1.14s]
01:28:20  38 of 95 START test unique_stg_metrics_unique_id ............................... [RUN]
01:28:20  25 of 95 OK loaded seed file snadbox.test_fct_unused_sources ................... [INSERT 1 in 3.63s]
01:28:20  39 of 95 START test not_null_stg_metric_relationships_unique_id ................ [RUN]
01:28:21  32 of 95 PASS unique_stg_model_2_id ............................................ [PASS in 1.03s]
01:28:21  40 of 95 START test unique_stg_metric_relationships_unique_id .................. [RUN]
01:28:21  33 of 95 PASS not_null_stg_exposure_relationships_unique_id .................... [PASS in 1.20s]
01:28:21  41 of 95 START test not_null_stg_node_relationships_unique_id .................. [RUN]
01:28:21  35 of 95 PASS not_null_stg_exposures_unique_id ................................. [PASS in 1.21s]
01:28:21  42 of 95 START test unique_stg_node_relationships_unique_id .................... [RUN]
01:28:21  37 of 95 PASS not_null_stg_metrics_unique_id ................................... [PASS in 1.21s]
01:28:21  43 of 95 START test not_null_stg_nodes_unique_id ............................... [RUN]
01:28:21  39 of 95 PASS not_null_stg_metric_relationships_unique_id ...................... [PASS in 1.11s]
01:28:21  44 of 95 START test unique_stg_nodes_unique_id ................................. [RUN]
01:28:21  38 of 95 PASS unique_stg_metrics_unique_id ..................................... [PASS in 1.35s]
01:28:21  45 of 95 START test not_null_stg_sources_unique_id ............................. [RUN]
01:28:22  34 of 95 PASS unique_stg_exposure_relationships_unique_id ...................... [PASS in 1.52s]
01:28:22  46 of 95 START test unique_stg_sources_unique_id ............................... [RUN]
01:28:22  36 of 95 PASS unique_stg_exposures_unique_id ................................... [PASS in 1.49s]
01:28:22  47 of 95 START test not_null_int_model_4_id .................................... [RUN]
01:28:22  40 of 95 PASS unique_stg_metric_relationships_unique_id ........................ [PASS in 1.33s]
01:28:22  48 of 95 START test unique_int_model_4_id ...................................... [RUN]
01:28:22  41 of 95 PASS not_null_stg_node_relationships_unique_id ........................ [PASS in 1.15s]
01:28:22  49 of 95 START test not_null_stg_model_4_id .................................... [RUN]
01:28:22  43 of 95 PASS not_null_stg_nodes_unique_id ..................................... [PASS in 1.09s]
01:28:22  50 of 95 START test unique_stg_model_4_id ...................................... [RUN]
01:28:22  47 of 95 PASS not_null_int_model_4_id .......................................... [PASS in 0.85s]
01:28:23  42 of 95 PASS unique_stg_node_relationships_unique_id .......................... [PASS in 1.22s]
01:28:23  44 of 95 PASS unique_stg_nodes_unique_id ....................................... [PASS in 1.23s]
01:28:23  45 of 95 PASS not_null_stg_sources_unique_id ................................... [PASS in 1.27s]
01:28:23  46 of 95 PASS unique_stg_sources_unique_id ..................................... [PASS in 1.22s]
01:28:23  51 of 95 START table model snadbox.int_all_graph_resources ..................... [RUN]
01:28:23  49 of 95 PASS not_null_stg_model_4_id .......................................... [PASS in 0.89s]
01:28:23  48 of 95 PASS unique_int_model_4_id ............................................ [PASS in 0.98s]
01:28:23  50 of 95 PASS unique_stg_model_4_id ............................................ [PASS in 1.02s]
01:28:27  51 of 95 OK created table model snadbox.int_all_graph_resources ................ [CREATE TABLE (85.0 rows, 29.8 KB processed) in 3.79s]
01:28:27  52 of 95 START test not_null_int_all_graph_resources_resource_id ............... [RUN]
01:28:27  53 of 95 START test unique_int_all_graph_resources_resource_id ................. [RUN]
01:28:28  53 of 95 PASS unique_int_all_graph_resources_resource_id ....................... [PASS in 1.24s]
01:28:28  52 of 95 PASS not_null_int_all_graph_resources_resource_id ..................... [PASS in 1.25s]
01:28:28  54 of 95 START view model snadbox.fct_documentation_coverage ................... [RUN]
01:28:28  55 of 95 START view model snadbox.fct_model_naming_conventions ................. [RUN]
01:28:28  56 of 95 START view model snadbox.fct_source_directories ....................... [RUN]
01:28:28  57 of 95 START view model snadbox.fct_undocumented_models ...................... [RUN]
01:28:28  58 of 95 START table model snadbox.int_direct_relationships .................... [RUN]
01:28:29  56 of 95 OK created view model snadbox.fct_source_directories .................. [OK in 0.83s]
01:28:29  59 of 95 START test equality_fct_source_directories ............................ [RUN]
01:28:29  57 of 95 OK created view model snadbox.fct_undocumented_models ................. [OK in 0.88s]
01:28:29  55 of 95 OK created view model snadbox.fct_model_naming_conventions ............ [OK in 0.89s]
01:28:29  60 of 95 START test equality_fct_undocumented_models ........................... [RUN]
01:28:29  61 of 95 START test equality_fct_model_naming_conventions ...................... [RUN]
01:28:29  54 of 95 OK created view model snadbox.fct_documentation_coverage .............. [OK in 0.95s]
01:28:29  62 of 95 START test equality_fct_documentation_coverage ........................ [RUN]
01:28:30  59 of 95 PASS equality_fct_source_directories .................................. [PASS in 1.79s]
01:28:30  58 of 95 OK created table model snadbox.int_direct_relationships ............... [CREATE TABLE (106.0 rows, 33.8 KB processed) in 2.67s]
01:28:30  63 of 95 START test not_null_int_direct_relationships_unique_id ................ [RUN]
01:28:30  64 of 95 START test unique_int_direct_relationships_unique_id .................. [RUN]
01:28:31  61 of 95 PASS equality_fct_model_naming_conventions ............................ [PASS in 1.83s]
01:28:31  60 of 95 PASS equality_fct_undocumented_models ................................. [PASS in 1.87s]
01:28:31  62 of 95 PASS equality_fct_documentation_coverage .............................. [PASS in 1.90s]
01:28:32  63 of 95 PASS not_null_int_direct_relationships_unique_id ...................... [PASS in 1.02s]
01:28:32  64 of 95 PASS unique_int_direct_relationships_unique_id ........................ [PASS in 1.19s]
01:28:32  65 of 95 START view model snadbox.fct_test_directories ......................... [RUN]
01:28:32  66 of 95 START view model snadbox.int_all_dag_relationships .................... [RUN]
01:28:32  67 of 95 START view model snadbox.int_model_test_summary ....................... [RUN]
01:28:33  67 of 95 OK created view model snadbox.int_model_test_summary .................. [OK in 0.81s]
01:28:33  65 of 95 OK created view model snadbox.fct_test_directories .................... [OK in 0.88s]
01:28:33  68 of 95 START test not_null_int_model_test_summary_resource_name .............. [RUN]
01:28:33  69 of 95 START test unique_int_model_test_summary_resource_name ................ [RUN]
01:28:33  70 of 95 START test equality_fct_test_directories .............................. [RUN]
01:28:33  66 of 95 OK created view model snadbox.int_all_dag_relationships ............... [OK in 0.89s]
01:28:33  71 of 95 START test not_null_int_all_dag_relationships_path .................... [RUN]
01:28:34  68 of 95 PASS not_null_int_model_test_summary_resource_name .................... [PASS in 1.70s]
01:28:34  69 of 95 PASS unique_int_model_test_summary_resource_name ...................... [PASS in 1.75s]
01:28:34  72 of 95 START view model snadbox.fct_missing_primary_key_tests ................ [RUN]
01:28:34  73 of 95 START view model snadbox.fct_test_coverage ............................ [RUN]
01:28:35  70 of 95 PASS equality_fct_test_directories .................................... [PASS in 1.94s]
01:28:35  73 of 95 OK created view model snadbox.fct_test_coverage ....................... [OK in 0.77s]
01:28:35  74 of 95 START test equality_fct_test_coverage ................................. [RUN]
01:28:35  72 of 95 OK created view model snadbox.fct_missing_primary_key_tests ........... [OK in 0.85s]
01:28:35  75 of 95 START test equality_fct_missing_primary_key_tests ..................... [RUN]
01:28:36  71 of 95 PASS not_null_int_all_dag_relationships_path .......................... [PASS in 3.41s]
01:28:36  76 of 95 START table model snadbox.fct_direct_join_to_source ................... [RUN]
01:28:36  77 of 95 START table model snadbox.fct_marts_or_intermediate_dependent_on_source  [RUN]
01:28:36  78 of 95 START view model snadbox.fct_model_directories ........................ [RUN]
01:28:36  79 of 95 START table model snadbox.fct_model_fanout ............................ [RUN]
01:28:36  80 of 95 START table model snadbox.fct_multiple_sources_joined ................. [RUN]
01:28:36  81 of 95 START table model snadbox.fct_rejoining_of_upstream_concepts .......... [RUN]
01:28:37  78 of 95 OK created view model snadbox.fct_model_directories ................... [OK in 0.89s]
01:28:37  82 of 95 START table model snadbox.fct_root_models ............................. [RUN]
01:28:38  74 of 95 PASS equality_fct_test_coverage ....................................... [PASS in 2.64s]
01:28:38  83 of 95 START table model snadbox.fct_source_fanout ........................... [RUN]
01:28:38  75 of 95 PASS equality_fct_missing_primary_key_tests ........................... [PASS in 2.91s]
01:28:38  84 of 95 START table model snadbox.fct_staging_dependent_on_marts_or_intermediate  [RUN]
01:28:39  77 of 95 OK created table model snadbox.fct_marts_or_intermediate_dependent_on_source  [CREATE TABLE (2.0 rows, 26.7 KB processed) in 2.78s]
01:28:39  85 of 95 START table model snadbox.fct_staging_dependent_on_staging ............ [RUN]
01:28:39  80 of 95 OK created table model snadbox.fct_multiple_sources_joined ............ [CREATE TABLE (1.0 rows, 26.6 KB processed) in 2.88s]
01:28:39  86 of 95 START table model snadbox.fct_unused_sources .......................... [RUN]
01:28:39  76 of 95 OK created table model snadbox.fct_direct_join_to_source .............. [CREATE TABLE (2.0 rows, 33.6 KB processed) in 3.36s]
01:28:39  87 of 95 START test equality_fct_model_directories ............................. [RUN]
01:28:40  83 of 95 OK created table model snadbox.fct_source_fanout ...................... [CREATE TABLE (2.0 rows, 26.6 KB processed) in 2.70s]
01:28:40  88 of 95 START test equality_fct_multiple_sources_joined ....................... [RUN]
01:28:41  79 of 95 OK created table model snadbox.fct_model_fanout ....................... [CREATE TABLE (1.0 rows, 26.6 KB processed) in 4.51s]
01:28:41  89 of 95 START test equality_fct_direct_join_to_source ......................... [RUN]
01:28:41  84 of 95 OK created table model snadbox.fct_staging_dependent_on_marts_or_intermediate  [CREATE TABLE (1.0 rows, 26.7 KB processed) in 2.76s]
01:28:41  90 of 95 START test equality_fct_source_fanout ................................. [RUN]
01:28:41  82 of 95 OK created table model snadbox.fct_root_models ........................ [CREATE TABLE (1.0 rows, 26.6 KB processed) in 3.99s]
01:28:41  91 of 95 START test equality_fct_model_fanout .................................. [RUN]
01:28:41  81 of 95 OK created table model snadbox.fct_rejoining_of_upstream_concepts ..... [CREATE TABLE (1.0 rows, 26.6 KB processed) in 5.25s]
01:28:41  92 of 95 START test equality_fct_root_models ................................... [RUN]
01:28:42  85 of 95 OK created table model snadbox.fct_staging_dependent_on_staging ....... [CREATE TABLE (1.0 rows, 26.7 KB processed) in 3.00s]
01:28:42  93 of 95 START test equality_fct_rejoining_of_upstream_concepts ................ [RUN]
01:28:42  89 of 95 PASS equality_fct_direct_join_to_source ............................... [PASS in 1.58s]
01:28:42  94 of 95 START test equality_fct_staging_dependent_on_staging .................. [RUN]
01:28:42  88 of 95 PASS equality_fct_multiple_sources_joined ............................. [PASS in 1.76s]
01:28:42  90 of 95 PASS equality_fct_source_fanout ....................................... [PASS in 1.62s]
01:28:42  91 of 95 PASS equality_fct_model_fanout ........................................ [PASS in 1.55s]
01:28:43  92 of 95 PASS equality_fct_root_models ......................................... [PASS in 1.74s]
01:28:43  86 of 95 OK created table model snadbox.fct_unused_sources ..................... [CREATE TABLE (1.0 rows, 26.6 KB processed) in 4.48s]
01:28:43  95 of 95 START test equality_fct_unused_sources ................................ [RUN]
01:28:43  93 of 95 PASS equality_fct_rejoining_of_upstream_concepts ...................... [PASS in 1.64s]
01:28:44  94 of 95 PASS equality_fct_staging_dependent_on_staging ........................ [PASS in 1.45s]
01:28:45  95 of 95 PASS equality_fct_unused_sources ...................................... [PASS in 1.59s]
01:28:46  87 of 95 PASS equality_fct_model_directories ................................... [PASS in 6.38s]
01:28:46  
01:28:46  Finished running 12 view models, 19 table models, 16 seeds, 48 tests in 38.02s.
01:28:46  
01:28:46  Completed successfully
01:28:46  
01:28:46  Done. PASS=95 WARN=0 ERROR=0 SKIP=0 TOTAL=95

```

## Checklist
- [x] I have verified that these changes work locally on the following warehouses (Note: it's okay if you do not have access to all warehouses, this helps us understand what has been covered)
    - [x] BigQuery
    - [ ] Postgres
    - [ ] Redshift
    - [ ] Snowflake
- [ ] I have updated the README.md (if applicable)
- [ ] I have added tests & descriptions to my models (and macros if applicable)
- [ ] I have added an entry to CHANGELOG.md

### CODE REVIEW

File: dbt_project.yml [hunk 1]
```diff
@@ -33,10 +33,7 @@ models:
         +materialized: table
     staging:
       graph:
-        stg_nodes:
-          +materialized: table
-        stg_node_relationships:
-          +materialized: table
+        +materialized: table
```
Comment (b-per) [inline dbt_project.yml:36]: This change seems to be breaking Redshift. Is there a reason why we would want all models under staging/graph to materialized as tables?

**Review by b-per — COMMENTED** [review]
I created #146 to trigger the CI check (@dave-connors-3  we need to activate CircleCI for forks to allow contributors to trigger CI checks)

It looks good but there is an error with Redshift now when trying to create a table with booleans. This error didn't happen before this PR. I left a comment in the code.

**Comment by b-per** [comment]
I have spent some time debugging it but I just don't understand why `stg_nodes` (which follows the same logic about casting a column to boolean is not failing materialized as a table when the other models are...)

File: dbt_project.yml (hunk 1 above)
Comment (VasiliiSurov) [reply dbt_project.yml:36]: These all are logically identical entities that represent different sub-types of a node so for me it totally make sense to have them all as tables, but if this change is causing issues you can revert it. Definitely out of scope of this bug

**Comment by VasiliiSurov** [comment]
> I have spent some time debugging it but I just don't understand why `stg_nodes` (which follows the same logic about casting a column to boolean is not failing materialized as a table when the other models are...)
I've never worked with Redshift but here is what I found https://stackoverflow.com/a/45799072
Looks like a broader issue with type casting for Redshift

**Comment by VasiliiSurov** [comment]
> It looks like we'll also need to update the documentation for this macro in `models/staging/graph/graph.yml`

done

File: models/staging/graph/graph.yml [hunk 2]
```diff
@@ -6,7 +6,12 @@ macros:
     arguments:
       - name: column_names
```
Comment (graciegoheen) [inline models/staging/graph/graph.yml:7]: I believe this argument name is now `columns`

Comment (VasiliiSurov) [reply models/staging/graph/graph.yml]: you're right, corrected

**Comment by b-per** [comment]
Hi @VasiliiSurov ! 

Thanks for the work on this one. We discussed internally and we are ok with merging the code with this approach. Can you please just fix the git conflict (due to new code having been pushed to `main`)? I will approve the PR and merge it just after.

(FYI, I created the branch https://github.com/dbt-labs/dbt-project-evaluator/pull/157 to check the CI which passed)

**Pull request merged**